components/brave_wallet/browser/zcash/zcash_keyring.cc

```diff
@@ -84,4 +186,24 @@ std::optional<std::vector<uint8_t>> ZCashKeyring::SignMessage(
   return hd_key->SignDer(message);
 }
 
+void ZCashKeyring::ConstructRootHDKey(const std::vector<uint8_t>& seed,
+                                      const std::string& hd_path) {
+  if (!seed.empty()) {
+    if (auto master_key = HDKey::GenerateFromSeed(seed)) {
```

Comment:
It would be cleaner by just call `HDKeyring::ConstructRootHDKey` here and check `(!seed.empty() && IsZCashShieldedEnabled()` to construct extended spending key.